### PR TITLE
SDL Support

### PR DESCRIPTION
**Pull request by benwilson512**
- [x] Absinthe.Schema.Notation => Blueprint Tree
- [x] SDL schema => Blueprint Tree
- [x] Blueprint Tree => compiled / usable schema

This branch is wildly experimental right now, do not try to use.

### CODE REVIEW

**Comment by benwilson512** [comment]
Unfortunately, the current implementation has the compilation time issue we ran into with early versions of Absinthe. The experimental implementation does:

```
@absinthe_blueprint Notation.put_attrs(@absinthe_blueprint, ...)
```

For ~200 line schemas it takes ~2 seconds to compile. Not great, but workable. The problem is it's non linear. A schema twice the size at ~400 lines takes 12 seconds to compile. 700 lines takes 26 seconds, and 1000 lines takes 45 seconds.

Basically, we can't do stuff in the module body. This may be OK though, since we're doing things differently this time. If we ignore the problem of anonymous functions for a second (I've got an idea for those) I think we can simply push values into an ets table. Then we do a single `@absinthe_blueprint` with the blob at the bottom.

File: lib/absinthe/phase/schema/build.ex [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    body = [
+      types,
+      quote do
+        def __absinthe_type__(_) do
+          nil
+        end
+      end
+    ]
+
+    Module.create(module_name, body, Macro.Env.location(__ENV__))
```
Comment (benwilson512) [inline lib/absinthe/phase/schema/build.ex]: @bruce the very beginning of blueprint => compiled schema work.

**Comment by collegeimprovements** [comment]
Thanks a lot @benwilson512 for SDL features. What can we expect? Does it mean it will include directive like prisma in nodejs world ?

**Comment by benwilson512** [comment]
It will yes.

**Comment by bruce** [comment]
@benwilson512 Is it fair to say this should be closed in favor of `switch-to-new-schema`?

**Comment by benwilson512** [comment]
The only remaining functionality item at this point I think is

- [ ] Subscription triggers

Then of course there's all the validation rules we need to recreate, and the error system.

**Comment by benwilson512** [comment]
OK. I'm gonna merge this in at this point, everything is fully functional. The error system still needs to happen, but I'll handle that in another PR. The next major task is actually over in Absinthe.Relay to make sure that we can get it to work with all of these changes.

**Comment by binaryseed** [comment]
closes #222
